pkg/background/update_request_controller.go

```diff
@@ -178,39 +175,30 @@ func (c *Controller) syncUpdateRequest(key string) error {
 	defer func() {
 		logger.V(4).Info("completed sync update request", "key", key, "processingTime", time.Since(startTime).String())
 	}()
-
 	_, urName, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
 		return err
 	}
-
-	ur, err := c.urLister.Get(urName)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil
+	if ur, err := c.urLister.Get(urName); err != nil {
+		return err
```

Comment:
Can we add details in return message?
```
return fmt.Errorf("failed to fetch update request %s: %v", key, err)
```

Reply:
We can but remember that we need the real error in `handleErr` for some cases. I usually prefer to check and react to errors in a single place.

Reply:
Fixed.

Reply:
I agree! But sometimes I find it's not easy to locate the exact error without adding a detailed message.